File: app/src/main/java/net/suntrans/szxf/uiv2/activity/KongtiaoActivity.java
package net.suntrans.szxf.uiv2.activity;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import net.suntrans.looney.widgets.IosAlertDialog;
import net.suntrans.szxf.App;
import net.suntrans.szxf.R;
import net.suntrans.szxf.activity.BasedActivity;
import net.suntrans.szxf.bean.RespondBody;
import net.suntrans.szxf.databinding.ActivityEnergyMoniBinding;
import net.suntrans.szxf.rx.BaseSubscriber;
import net.suntrans.szxf.uiv2.bean.AirCmd;
import net.suntrans.szxf.uiv2.bean.Monitor;
import net.suntrans.szxf.uiv2.fragment.EnvListFragment;
import net.suntrans.szxf.uiv2.fragment.KongtiaoFragment;
import net.suntrans.szxf.uiv2.fragment.KongtiaoFragment_type1;

import java.util.List;

/**
 * Created by Looney on 2017/11/22.
 * Des:
 */

public class KongtiaoActivity extends BasedActivity {

    private ActivityEnergyMoniBinding binding;
    private String channel_id;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_energy_moni);
        binding.back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        TextView txTitle = (TextView) findViewById(R.id.title);
        txTitle.setText("空调控制");

        findViewById(R.id.back)
                .setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        finish();
                    }
                });

        channel_id = getIntent().getStringExtra("channel_id");


        getData(channel_id);

    }


    private void getData(String channel_ids) {
        addSubscription(api.getTotalCmd(channel_ids), new BaseSubscriber<RespondBody<List<AirCmd>>>(this) {
            @Override
            public void onNext(RespondBody<List<AirCmd>> body) {

                if (body.data != null && body.data.size() > 1) {
                    if ("0".equals(body.data.get(0).air_type)) {
                        KongtiaoFragment fragment = KongtiaoFragment.newInstance(channel_id);
                        getSupportFragmentManager().beginTransaction()
                                .replace(R.id.content, fragment)
                                .commit();
                    } else if ("1".equals(body.data.get(0).air_type)){

                        KongtiaoFragment_type1 fragment = KongtiaoFragment_type1.newInstance(channel_id);
                        getSupportFragmentManager().beginTransaction()
                                .replace(R.id.content, fragment)
                                .commit();
                    }else {
                        new IosAlertDialog(KongtiaoActivity.this)
                                .builder()
                                .setCancelable(false)
                                .setMsg("不支持的空调类型...")
                                .setPositiveButton(getResources().getString(R.string.close), new View.OnClickListener() {
                                    @Override
                                    public void onClick(View v) {
                                        finish();
                                    }
                                }).show();
                    }

                }else {
                    new IosAlertDialog(KongtiaoActivity.this)
                            .builder()
                            .setCancelable(false)
                            .setMsg("不支持的空调类型...")
                            .setPositiveButton(getResources().getString(R.string.close), new View.OnClickListener() {
                                @Override
                                public void onClick(View v) {
                                    finish();
                                }
                            }).show();
                }

            }

            @Override
            public void onError(Throwable e) {

                e.printStackTrace();

                super.onError(e);
            }
        });
    }


}

File: app/src/main/java/net/suntrans/szxf/uiv2/activity/KongtiaoActivity.kt
package net.suntrans.szxf.uiv2.activity

import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.View
import android.widget.TextView

import net.suntrans.looney.widgets.IosAlertDialog
import net.suntrans.szxf.R
import net.suntrans.szxf.activity.BasedActivity
import net.suntrans.szxf.bean.RespondBody
import net.suntrans.szxf.databinding.ActivityEnergyMoniBinding
import net.suntrans.szxf.rx.BaseSubscriber
import net.suntrans.szxf.uiv2.bean.AirCmd
import net.suntrans.szxf.uiv2.fragment.KongtiaoFragment

/**
 * Created by Looney on 2017/11/22.
 * Des:
 */

class KongtiaoActivity : BasedActivity() {

    private var binding: ActivityEnergyMoniBinding? = null
    private var channel_id: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_energy_moni)
        binding!!.back.setOnClickListener { finish() }

        val txTitle = findViewById(R.id.title) as TextView
        val title = intent.getStringExtra("title") + "空调"
        txTitle.text = title

        findViewById(R.id.back)
                .setOnClickListener { finish() }

        channel_id = intent.getStringExtra("channel_id")
        println("channel_id:"+channel_id)

        binding!!.rightSubTitle.visibility = View.VISIBLE
        binding!!.rightSubTitle
                .setText(R.string.sub_title_liandong)

        binding!!.rightSubTitle
                .setOnClickListener {
                    val intent = Intent(this@KongtiaoActivity, KtAutoActivity::class.java)
                    intent.putExtra("channel_id", channel_id)
                    intent.putExtra("title", title)
                    startActivity(intent)
                }


        val fragment = KongtiaoFragment.newInstance(channel_id)
        supportFragmentManager.beginTransaction()
                .replace(R.id.content, fragment)
                .commit()

    }


    private fun getData(channel_ids: String?) {
        addSubscription(api.getTotalCmd(channel_ids), object : BaseSubscriber<RespondBody<List<AirCmd>>>(this) {
            override fun onNext(body: RespondBody<List<AirCmd>>) {

                if (body.data != null && body.data.size > 1) {


                } else {
                    IosAlertDialog(this@KongtiaoActivity)
                            .builder()
                            .setCancelable(false)
                            .setMsg(getString(R.string.tips_kt_unsported))
                            .setPositiveButton(resources.getString(R.string.close)) { finish() }.show()
                }

            }

            override fun onError(e: Throwable) {

                e.printStackTrace()

                super.onError(e)
            }
        })
    }


}
